src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,34 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.UserAlreadyExistException;
+import core.basesyntax.exceptions.UsersAgeNotValidException;
+import core.basesyntax.exceptions.UsersPasswordNotValidException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null
+                || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new NullPointerException("You must fill all the fields!");
+        }
+        if (storageDao.get(user.getLogin()) == null) {
+            if (user.getAge() >= 18) {
```

Comment:
From common mistakes: Move all hardcoded values to constant fields and give them informative names. ALLOWED_AGE for example